pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1070,7 +1244,32 @@ def _parse_control_values(control_wires, control_values):
         else:
             raise ValueError("Alternative control values must be passed as a binary string.")
 
-        return control_int
+        padding_left = control_int * 2
+        padding_right = 2 ** (len(control_wires)+1) - 2 - padding_left
+        cx = block_diag(np.eye(padding_left), PauliX.compute_matrix(), np.eye(padding_right))
```

Comment:
The final matrix gets computed here from scratch in every call.

Reply:
:+1: